From f89a4e3da45753d012b3c40bfe9c8b56671e5306 Mon Sep 17 00:00:00 2001
From: You-Sheng Yang <vicamo.yang@canonical.com>
Date: Fri, 11 Oct 2024 01:28:38 +0800
Subject: [PATCH 6/6] media: ipu6: Fix compilation with kernels >= 6.12:
 Finally take no_llseek out

Accommodate to v6.12-rc1 commit cb787f4ac0c2 ("[tree-wide] finally take
no_llseek out").

Bug-Ubuntu: https://bugs.launchpad.net/bugs/2083996
Signed-off-by: You-Sheng Yang <vicamo.yang@canonical.com>
Signed-off-by: Hans de Goede <hdegoede@redhat.com>
---
 drivers/media/pci/intel/ipu-trace.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/media/pci/intel/ipu-trace.c b/drivers/media/pci/intel/ipu-trace.c
index 282ac659a..2dc458a1d 100644
--- a/drivers/media/pci/intel/ipu-trace.c
+++ b/drivers/media/pci/intel/ipu-trace.c
@@ -517,7 +517,9 @@ static const struct file_operations ipu_traceconf_fops = {
 	.release = traceconf_release,
 	.read = traceconf_read,
 	.write = traceconf_write,
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 12, 0)
 	.llseek = no_llseek,
+#endif
 };
 
 static void wptraceconf_dump(struct ipu_device *isp)
@@ -673,7 +675,9 @@ static const struct file_operations ipu_wptraceconf_fops = {
 	.release = wptraceconf_release,
 	.read = wptraceconf_read,
 	.write = wptraceconf_write,
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 12, 0)
 	.llseek = no_llseek,
+#endif
 };
 
 static int gettrace_open(struct inode *inode, struct file *file)
@@ -740,7 +744,9 @@ static const struct file_operations ipu_gettrace_fops = {
 	.release = gettrace_release,
 	.read = gettrace_read,
 	.write = gettrace_write,
+#if LINUX_VERSION_CODE < KERNEL_VERSION(6, 12, 0)
 	.llseek = no_llseek,
+#endif
 };
 
 int ipu_trace_init(struct ipu_device *isp, void __iomem *base,
-- 
2.47.0

